lib/taskana-core-test/src/test/java/acceptance/classification/query/QueryClassificationsFilteringAccTest.java

```diff
@@ -0,0 +1,724 @@
+import static pro.taskana.classification.api.ClassificationCustomField.CUSTOM_3;
+import static pro.taskana.classification.api.ClassificationCustomField.CUSTOM_4;
+import static pro.taskana.classification.api.ClassificationCustomField.CUSTOM_5;
+import static pro.taskana.classification.api.ClassificationCustomField.CUSTOM_6;
+import static pro.taskana.classification.api.ClassificationCustomField.CUSTOM_7;
+import static pro.taskana.classification.api.ClassificationCustomField.CUSTOM_8;
+import static pro.taskana.classification.api.ClassificationQueryColumnName.CREATED;
+import static pro.taskana.classification.api.ClassificationQueryColumnName.NAME;
+import static pro.taskana.classification.api.ClassificationQueryColumnName.TYPE;
+import static pro.taskana.classification.api.ClassificationQueryColumnName.VALID_IN_DOMAIN;
+import static pro.taskana.testapi.DefaultTestEntities.defaultTestClassification;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestFactory;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.function.ThrowingConsumer;
+
+import pro.taskana.classification.api.ClassificationCustomField;
+import pro.taskana.classification.api.ClassificationService;
+import pro.taskana.classification.api.models.Classification;
+import pro.taskana.classification.api.models.ClassificationSummary;
+import pro.taskana.common.api.TimeInterval;
+import pro.taskana.common.internal.util.Pair;
+import pro.taskana.testapi.TaskanaInject;
+import pro.taskana.testapi.TaskanaIntegrationTest;
+import pro.taskana.testapi.security.WithAccessId;
+
+/**
+ * The QueryClassificationsFilteringAccTest for all "query classification with filtering" scenarios.
+ *
+ * <p>Note: Instead of working with a clean database for each test, we create a scenario. In that
+ * scenario we use {@linkplain ClassificationCustomField#CUSTOM_1} to filter out all the other
+ * created {@linkplain Classification Classifications}. Therefore, all tests regarding {@linkplain
+ * ClassificationCustomField ClassificationCustomFields} are executed first.
+ *
+ * <p>The alternative is to create a "big" scenario as a test setup. But that would result to the
+ * same issues we have with the sql file.
+ */
+@TaskanaIntegrationTest
+@TestMethodOrder(value = MethodOrderer.OrderAnnotation.class)
+public class QueryClassificationsFilteringAccTest {
```

Comment:
Please remove the public modifier. What do you think about naming the class ClassificationQueryFilteringAccTest?